homeassistant/components/starline/binary_sensor.py

```diff
@@ -0,0 +1,71 @@
+"""Reads vehicle status from StarLine API."""
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice,
+    DEVICE_CLASS_OPENING,
+    DEVICE_CLASS_LOCK,
+    DEVICE_CLASS_PROBLEM,
+    DEVICE_CLASS_POWER,
+)
+from .account import StarlineAccount, StarlineDevice
+from .const import DOMAIN
+from .entity import StarlineEntity
+
+SENSOR_TYPES = {
+    "hbrake": [
+        "Hand Brake",
+        DEVICE_CLASS_POWER,
+        "mdi:car-brake-parking",
+        "mdi:car-brake-parking",
+    ],
+    "hood": ["Hood", DEVICE_CLASS_OPENING, "mdi:car", "mdi:car"],
+    "trunk": ["Trunk", DEVICE_CLASS_OPENING, "mdi:car-back", "mdi:car-back"],
+    "alarm": ["Alarm", DEVICE_CLASS_PROBLEM, "mdi:car-connected", "mdi:car"],
+    "door": ["Doors", DEVICE_CLASS_LOCK, "mdi:car-door", "mdi:car-door-lock"],
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the StarLine sensors."""
+    account: StarlineAccount = hass.data[DOMAIN][entry.entry_id]
+    entities = []
+    for device in account.api.devices.values():
+        for key, value in SENSOR_TYPES.items():
+            if key in device.car_state:
+                entities.append(StarlineSensor(account, device, key, *value))
+    async_add_entities(entities)
+    return True
+
+
+class StarlineSensor(StarlineEntity, BinarySensorDevice):
+    """Representation of a StarLine binary sensor."""
+
+    def __init__(
+        self,
+        account: StarlineAccount,
+        device: StarlineDevice,
+        key: str,
+        name: str,
+        device_class: str,
+        icon_on: str,
+        icon_off: str,
+    ):
+        """Constructor."""
+        super().__init__(account, device, key, name)
+        self._device_class = device_class
+        self._icon_on = icon_on
+        self._icon_off = icon_off
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon_on if self.is_on else self._icon_off
```

Comment:
If a device class exists, don't set manual an icon.

Reply:
Why not? I think, special icons for trunk or hood are good here.

Reply:
@MartinHjelmare please answer my question.
Why i can't use icons for doors? If it's really required, i think, it's better to remove device class and keep good icons here.

Reply:
The point of the device class is that the entity with a device class should be represented the same way in home assistant, ie use the same unit and the same icon.

If the feature of the device doesn't fit into a device class, we shouldn't use it.